crates/api_crud/src/custom_emoji/create.rs

```diff
@@ -0,0 +1,46 @@
+use crate::PerformCrud;
+use actix_web::web::Data;
+use lemmy_api_common::{
+  context::LemmyContext,
+  custom_emoji::{CreateCustomEmoji, CustomEmojiResponse},
+  utils::{get_local_user_view_from_jwt, is_admin},
+};
+use lemmy_db_schema::source::local_site::LocalSite;
+use lemmy_db_views::structs::{CustomEmojiView, CustomEmojiViewForm};
+use lemmy_utils::{error::LemmyError, ConnectionId};
+
+#[async_trait::async_trait(?Send)]
+impl PerformCrud for CreateCustomEmoji {
+  type Response = CustomEmojiResponse;
+
+  #[tracing::instrument(skip(self, context, _websocket_id))]
+  async fn perform(
+    &self,
+    context: &Data<LemmyContext>,
+    _websocket_id: Option<ConnectionId>,
+  ) -> Result<CustomEmojiResponse, LemmyError> {
+    let data: &CreateCustomEmoji = self;
+    let local_user_view =
+      get_local_user_view_from_jwt(&data.auth, context.pool(), context.secret()).await?;
+
+    let local_site = LocalSite::read(context.pool()).await?;
+    // Make sure user is an admin
+    is_admin(&local_user_view)?;
+
+    let form = data.clone();
+    let res = CustomEmojiView::create(
```

Comment:
All the other api examples show how this is done. A file in `db_schema/impls` has all the insert / creation helper methods, follow their example.

Reply:
I couldn't find a good example of this elsewhere, where its a one-to-many situation, where custom_emoji and custom_emoji_keywords need to be saved together. 

I put CustomEmojiView (and form) in the db_views crate, but I didn't think it was entirely correct. Could you give me some guidance on how you want this structured? 

I know I should have CustomEmoji and CustomEmojiKeyword in db_schema. Should I have a third class which is a composite? If so, where should it go? Diesel doesn't allow for structured db classes (where I can have a vec<keywords> on the parent class, so I feel like i need a new struct to handle that and compose the CustomEmoji with its keywords.